dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/security/apikey/ApiTokenServiceImplTest.java

```diff
@@ -215,4 +219,33 @@ private void switchToOtherUser()
         final User otherUser = createUserWithAuth( "otherUser" );
         injectSecurityContext( otherUser );
     }
+
+    @Test
+    void testValidateChecksums()
+    {
+        char[] token = ApiTokenServiceImpl.generatePlainTextToken( ApiTokenType.PERSONAL_ACCESS_TOKEN_V1 );
+        assertTrue( validateChecksum( token ) );
+    }
+
+    @Test
+    void testHashingToken()
+    {
+        char[] token = ApiTokenServiceImpl.generatePlainTextToken( ApiTokenType.PERSONAL_ACCESS_TOKEN_V1 );
+        String hashedToken = hashToken( token );
+        assertTrue( isValidSHA256( hashedToken ) );
+    }
+
+    private static boolean isValidSHA256( String s )
```

Comment:
if this logic is used to assert in a test, should it actually be in the src code itself (not in a test) with it's own test?
for instance, this is just validating test instances of a valid hash. Should this also be done in the code?